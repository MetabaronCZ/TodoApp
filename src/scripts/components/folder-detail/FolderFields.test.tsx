import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';

import * as ItemList from 'components/common/ItemList';
import { FolderFields } from 'components/folder-detail/FolderFields';

import { FolderData } from 'models/Folder';
import { withMockedProviders } from 'test/component';

const testData: FolderData = {
  title: 'Edited folder',
};

describe('components/folder-detail/FolderFields', () => {
  it('should render correctly', () => {
    const { container } = render(
      withMockedProviders(
        <FolderFields fields={testData} onChange={() => null} />,
      ),
    );
    expect(container).toMatchSnapshot();
  });

  it('should render props', () => {
    const itemList = jest.spyOn(ItemList, 'ItemList');
    expect(itemList).toHaveBeenCalledTimes(0);

    const tree = render(
      withMockedProviders(
        <FolderFields
          fields={testData}
          errors={{ title: 'Title field error' }}
          loading
          onChange={() => null}
        />,
      ),
    );
    expect(tree.container).toMatchSnapshot();

    expect(itemList).toHaveBeenCalledTimes(1);
    expect(itemList.mock.calls[0][0].loading).toEqual(true);

    const input = tree.container.querySelector('input');
    expect(input).toBeInTheDocument();
    expect(input?.value).toEqual('Edited folder');

    expect(tree.getByText('Title field error')).toBeInTheDocument();
  });

  it('should call onChange', () => {
    const onChange = jest.fn();

    const tree = render(
      withMockedProviders(
        <FolderFields fields={testData} onChange={onChange} />,
      ),
    );
    expect(onChange).toHaveBeenCalledTimes(0);

    const input = tree.container.querySelector('input');
    expect(input).toBeInTheDocument();

    // trigger onChange event
    if (input) {
      fireEvent.change(input, { target: { value: 'New title' } });
    }
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('title', 'New title');
  });
});
