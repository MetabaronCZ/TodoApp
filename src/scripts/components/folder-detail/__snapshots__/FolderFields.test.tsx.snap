// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/folder-detail/FolderFields should render correctly 1`] = `
.c0 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-bottom: 1px solid #c0c0c0;
}

.c1:last-child {
  border-bottom: none;
}

.c4 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #222;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  border: none;
  outline: none;
}

.c7:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c7:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c7:disabled {
  color: #c0c0c0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  min-height: 30px;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="form-field-0"
          >
            Název složky:
          </label>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              class="c7"
              id="form-field-0"
              maxlength="80"
              type="text"
              value="Edited folder"
            />
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`components/folder-detail/FolderFields should render props 1`] = `
.c0 {
  opacity: 0.5;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border-bottom: 1px solid #c0c0c0;
}

.c1:last-child {
  border-bottom: none;
}

.c4 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #222;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 100%;
  padding: 5px 10px;
  background: #fff;
  color: #c22;
  border: none;
  outline: none;
}

.c7:hover:not(:disabled) {
  background-color: #fcfcfc;
}

.c7:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c7:disabled {
  color: #c0c0c0;
}

.c8 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  padding: 0 10px;
  color: #fff;
  background: #c22;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100px;
  min-height: 30px;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <label
            class="c4"
            for="form-field-1"
          >
            Název složky:
          </label>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              class="c7"
              id="form-field-1"
              maxlength="80"
              type="text"
              value="Edited folder"
            />
            <div
              class="c8"
            >
              Title field error
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
